### ISSUE TITLE
Confidence pt vs ONNX (predict vs OpenCV)

### ISSUE THREAD

**Issue by knoppmyth** [ultralytics/ultralytics#1247]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I created a classification model which I then export to ONNX use with OpenCV. When running the model with OpenCV, the confidence show I was getting was really low compared to running predict against the .pt. So I decided to run predict against the ONNX.

The pt:
`[root@03dd6383d43a yolov8]# yolo classify predict model=runs/classify/train2/weights/best.pt source=../data/face2.jpg
Ultralytics YOLOv8.0.48 🚀 Python-3.10.9 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 2070 SUPER, 7973MiB)
YOLOv8s-cls summary (fused): 73 layers, 5084167 parameters, 0 gradients, 12.5 GFLOPs

image 1/1 /usr/src/data/face2.jpg: 64x64 neutral 0.77, sad 0.12, fear 0.06, angry 0.05, surprise 0.01, 2.8ms
Speed: 0.2ms preprocess, 2.8ms inference, 0.1ms postprocess per image at shape (1, 3, 64, 64)`

ONNX:
`[root@03dd6383d43a yolov8]# yolo classify predict model=runs/classify/train2/weights/best.onnx source=../data/face2.jpg imgsz=48
Ultralytics YOLOv8.0.48 🚀 Python-3.10.9 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 2070 SUPER, 7973MiB)
Loading runs/classify/train2/weights/best.onnx for ONNX Runtime inference...

image 1/1 /usr/src/data/face2.jpg: 48x48 neutral 0.50, sad 0.32, fear 0.10, angry 0.06, happy 0.02, 1.6ms
Speed: 0.2ms preprocess, 1.6ms inference, 0.0ms postprocess per image at shape (1, 3, 48, 48)`

Even here, the ONNX confidence is lower. Item of note, when training, the image sizes are 48x48. So when exporting, I specified --imgsz=48,48. The pt model requires the size in multiples of 32 so it used a size of 64x64. Could this have an affect on the confidence?

To answer my own question, yes it does have an effect. If I change export imgsz to 64 and run predict again, the score is the same as with the pt. However the confidence score I'm getting from OpenCV is still lower.

Here is the code I'm using:
`import cv2
import numpy as np
 
onnx_model_path = "best.onnx"
sample_image = "face2.jpg"
labels = ['ANGRY', 'DISGUST', 'FEAR', 'HAPPY', 'NEUTRAL', 'SAD', 'SURPRISE' ]
net =  cv2.dnn.readNetFromONNX(onnx_model_path) 
image = cv2.imread(sample_image)
blob = cv2.dnn.blobFromImage(image, 1.0 / 255, (64, 64),(0, 0, 0), swapRB=True, crop=False)
net.setInput(blob)
preds = net.forward()
biggest_pred_index = np.array(preds)[0].argmax()
confidence = preds[0][biggest_pred_index]
print ("Predicted class:",biggest_pred_index)
print("The class",biggest_pred_index, "correspond to", labels[biggest_pred_index], "with a confidence: {:.2f}".format(confidence.item()))`

Can anyone provide any insight as to why the confidence would be so much lower with OpenCV?

**Comment by glenn-jocher** [comment]
@knoppmyth CI export benchmarks show no loss in accuracy in exported model formats when trained and exported at the same image sizes:
https://github.com/ultralytics/ultralytics/actions/runs/4328314139/jobs/7557872756

<img width="611" alt="Screenshot 2023-03-04 at 12 02 30" src="https://user-images.githubusercontent.com/26833433/222926388-2b2e63c3-a30c-47d2-800b-ed4231cf1335.png">

**Comment by knoppmyth** [comment]
@glenn-jocher Thanks for the reply. Yes, after changing the onnx model input to 64x64, I saw I was getting the same confidence score. My question/concern is when I use that model with OpenCV, the confidence that is returned is far less than with the PyTorch model or the ONNX model running with "yolo predict". While I saw seeing 70+ with "predict", with OpenCV, I was seeing 40+. Do you have any thoughts on that?

Thanks again.
